package/MDAnalysis/coordinates/base.py

```diff
@@ -980,6 +982,79 @@ def __repr__(self):
+            coordinates from. Defaults to ``None``, in which case the full set of
+            coordinate data is returned.
+        start :  int (optional)
+             Begin reading the trajectory at frame index `start` (where 0 is the index
+             of the first frame in the trajectory); the default ``None`` starts
+             at the beginning.
+        stop : int (optional)
+             End reading the trajectory at frame index `stop`-1, i.e, `stop` is excluded.
+             The trajectory is read to the end with the default ``None``.
+        step : int (optional)
+             Step size for reading; the default ``None`` is equivalent to 1 and means to
+             read every frame.
+        order : str (optional)
+            the order/shape of the return data array, corresponding
+            to (a)tom, (f)rame, (c)oordinates all six combinations
+            of 'a', 'f', 'c' are allowed ie "fac" - return array
+            where the shape is (frame, number of atoms,
+            coordinates)
+        See Also
+        --------
+        MDAnalysis.coordinates.memory
+        .. versionadded:: 2.4.0
+        """
+        start, stop, step = self.check_slice_indices(start, stop, step)
+        nframes = len(range(start, stop, step))
+
+        if sorted(order) != sorted(self._default_timeseries_order):
+            raise ValueError("The order of the returned coordinate array must"
+                            " be a permutation of `fac`.")
+        if asel is not None:
+            if len(asel) == 0:
+                raise NoDataError(
+                    "Timeseries requires at least one atom to analyze")
+            atom_numbers = asel.indices
+            natoms = len(atom_numbers)
+        else:
+            natoms = self.n_atoms
+            atom_numbers = np.arange(natoms)
+
+        # allocate output array in 'fac' (_default_timeseries_order) order
+        coordinates = np.empty((nframes, natoms, 3), dtype=np.float32)
+        for i, ts in enumerate(self[start:stop:step]):
+            coordinates[i, :] = ts.positions[atom_numbers]
+
+        # switch axes around
+        if order == self._default_timeseries_order:
+            pass
+        elif order[0] == self._default_timeseries_order[0]:
+            coordinates = np.swapaxes(coordinates, 1, 2)
+        elif order[1] == self._default_timeseries_order[1]:
+            coordinates = np.swapaxes(coordinates, 0, 2)
+        elif order[2] == self._default_timeseries_order[2]:
+            coordinates = np.swapaxes(coordinates, 0, 1)
+        elif order[0] == self._default_timeseries_order[1]:
+            coordinates = np.swapaxes(coordinates, 1, 0)
+            coordinates = np.swapaxes(coordinates, 1, 2)
+        elif order[0] == self._default_timeseries_order[2]:
+            coordinates = np.swapaxes(coordinates, 2, 0)
+            coordinates = np.swapaxes(coordinates, 1, 2)
+        else:
+            pass
```

Comment:
```suggestion
        default_order = 'fac'
        if order != default_order:
            try:
                newidx = [default_order.index(i) for i in order]
            except ValueError:
                raise ValueError(f"Unrecognized order key in {order}, "
                                 "must be permutation of 'fac'")

            try:
                coordinates = np.moveaxis(coordinates, [0, 1, 2], newidx)
            except ValueError:
                errmsg = ("Repeated or missing keys passed to argument "
                          f"`order`: {order}, each key must be used once")
                raise ValueError(errmsg)
```
That should take care of order checking + also avoids the whole if/else tree.